File: backend/tests/fixtures/callgraph-sample/UserController.java
package com.example.demo;

import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/users")
public class UserController {

    @GetMapping("/{id}")
    public User getUser(@PathVariable Long id) {
        return fetchUser(id);
    }

    private User fetchUser(Long id) {
        return new User(id, "Alice");
    }
}

class User {
    private Long id;
    private String name;

    public User(Long id, String name) {
        this.id = id;
        this.name = name;
    }
}

File: backend/tests/fixtures/callgraph-sample/UserController.kt
package com.example

import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/users")
class UserController {
    @GetMapping("/{id}")
    fun getUser(@PathVariable id: Long): String {
        return fetchUser(id)
    }

    private fun fetchUser(id: Long): String {
        return "user"
    }
}
